addons/stock/models/stock_quant.py

```diff
@@ -524,12 +524,16 @@ def _merge_quants(self):
         query = """WITH
                         dupes AS (
                             SELECT min(id) as to_update_quant_id,
-                                (array_agg(id ORDER BY id))[2:array_length(array_agg(id), 1)] as to_delete_quant_ids,
+                                (array_agg(id ORDER BY id)) as to_delete_quant_ids,
                                 SUM(reserved_quantity) as reserved_quantity,
                                 SUM(quantity) as quantity
                             FROM stock_quant
+                            WHERE id in (SELECT dups.id FROM
+                                (select id, ROW_NUMBER() OVER(
+                                    PARTITION BY product_id, company_id, location_id, lot_id, package_id, owner_id, in_date order by id asc
```

Comment:
You shouldn't add the order by id after the partition by. To ensure that the min id is never in the same set?